kahuna/public/js/edits/service.js

```diff
@@ -149,6 +149,17 @@ service.factory('editsService',
         return existingRequestPool.promise;
     }
 
+    function perform(resource, action, originalImage) {
+        const newRequest = resource.perform(action).
+              then(action => resource.get().then(edit => getSynced(originalImage, newImage => matches(edit, newImage))));
```

Comment:
@theefer - should an action return itself in the response, or the response it's modified. At the moment we're doing a mix of returning the resource with the `uri` of the action.